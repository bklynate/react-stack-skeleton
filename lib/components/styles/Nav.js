'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactcss = require('reactcss');

var _reactcss2 = _interopRequireDefault(_reactcss);

var _A = require('./A');

var _A2 = _interopRequireDefault(_A);

var _responsiveEnhancements = require('../../containers/responsiveEnhancements');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The nav bar for the website.
 */
function Nav(_ref) {
  var screenSize = _ref.screenSize;
  var items = _ref.items;

  var otherProps = _objectWithoutProperties(_ref, ['screenSize', 'items']);

  var styles = (0, _reactcss2.default)({
    'default': {
      ul: {
        listStyleType: 'none'
      },
      li: {
        float: 'left',
        width: 100,
        marginLeft: 25
      }
    },
    'screenSize-xs': {
      li: {
        float: 'initial',
        width: '90%',
        marginLeft: '5%',
        marginRight: '5%'
      }
    }
  }, { screenSize: screenSize });

  return _react2.default.createElement(
    'nav',
    otherProps,
    _react2.default.createElement(
      'ul',
      { style: styles.ul },
      items.map(function (item, index) {
        return _react2.default.createElement(
          'li',
          { style: styles.li, key: index },
          _react2.default.createElement(_A2.default, item)
        );
      })
    )
  );
}
exports.default = (0, _responsiveEnhancements.useScreenSize)(Nav);