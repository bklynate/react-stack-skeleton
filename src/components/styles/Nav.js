/* @flow */
import React from 'react';
import reactCSS from 'reactcss';

import A from './A';
import { useScreenSize } from '../../containers/responsiveEnhancements';

import type { NavItem } from '../../stores/navigation';

type Props = {
  screenSize: string,
  items: Array<NavItem>,
};

/**
 * The nav bar for the website.
 */
function Nav({ screenSize, items, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      ul: {
        listStyleType: 'none',
      },
      li: {
        float: 'left',
        width: 100,
        marginLeft: 25,
      },
    },
    'screenSize-xs': {
      li: {
        float: 'initial',
        width: '90%',
        marginLeft: '5%',
        marginRight: '5%',
      },
    },
  }, { screenSize });

  return (
    <nav {...otherProps}>
      <ul  style={styles.ul}>
      {items.map((item, index) =>
        <li style={styles.li} key={index}>
          <A {...item} />
        </li>
      )}
      </ul>
    </nav>
  );
}
export default useScreenSize(Nav);
