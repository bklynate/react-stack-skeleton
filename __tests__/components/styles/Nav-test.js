import React from 'react';
import renderer from 'react-test-renderer';

import Nav from '../../../src/components/styles/Nav';

jest.mock('../../../src/containers/responsiveEnhancements', () => ({
  useScreenSize: jest.fn((C) => (props) => <C screenSize="lg" {...props} />),
}));

test('The Nav component displays the given navigation items.', () => {
  const component = renderer.create(
    <Nav
      items={[
        { label: 'Home', to: '/' },
        { label: 'Internal Link', to: '/internal' },
        { label: 'External Link', href: 'https://google.com' },
      ]}
    />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
