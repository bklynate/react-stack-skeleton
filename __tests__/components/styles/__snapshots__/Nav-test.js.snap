exports[`test The Nav component displays the given navigation items. 1`] = `
<nav>
  <ul
    style={
      Object {
        "listStyleType": "none",
      }
    }>
    <li
      style={
        Object {
          "float": "left",
          "marginLeft": 25,
          "width": 100,
        }
      }>
      <a
        onClick={[Function]}
        style={Object {}}>
        Home
      </a>
    </li>
    <li
      style={
        Object {
          "float": "left",
          "marginLeft": 25,
          "width": 100,
        }
      }>
      <a
        onClick={[Function]}
        style={Object {}}>
        Internal Link
      </a>
    </li>
    <li
      style={
        Object {
          "float": "left",
          "marginLeft": 25,
          "width": 100,
        }
      }>
      <a
        href="https://google.com">
        External Link
      </a>
    </li>
  </ul>
</nav>
`;
